### DISCUSSION TITLE
Add option to change PR title and description

### DISCUSSION THREAD

**Discussion by dhruvkb** [BetaHuhn/repo-file-sync-action#78] [Ideas] [answered]
Thanks for this very thoughtful action. It's perfectly suited to our needs!

Many fields, such the PR description for example, have been hardcoded into the codebase and cannot be changed via any config parameter.
https://github.com/BetaHuhn/repo-file-sync-action/blob/fe89855ed70c22342e47a92e39d1c5d3b9e6d91e/src/git.js#L194

It'd be nice if we could edit these fields via the `sync.yml` config or via params to the action.
- PR title and description
- Commit messages

**Comment by BetaHuhn** [comment]
Hey @dhruvkb thanks for taking the time to create this discussion!

> Many fields, such the PR description for example, have been hardcoded into the codebase and cannot be changed via any config parameter

You can already change the [commit body](https://github.com/BetaHuhn/repo-file-sync-action#custom-commit-body), the commit/pr [prefix](https://github.com/BetaHuhn/repo-file-sync-action#%EF%B8%8F-action-inputs) and I have been thinking about using the message of original commit for sync commit (https://github.com/BetaHuhn/repo-file-sync-action/discussions/13), so this is definitely a valid request as well!

> It'd be nice if we could edit these fields via the sync.yml config or via params to the action.

We could add another action input like `PR_TITLE` and `PR_DESCRIPTION` similarly to the `COMMIT_BODY` option.

How would you imagine theses options to work? Would it overwrite the title and description completely or just change parts of it? For example the PR body contains all of the changed files so overwriting it completely would mean they wouldn't be included:
https://github.com/BetaHuhn/repo-file-sync-action/blob/fe89855ed70c22342e47a92e39d1c5d3b9e6d91e/src/git.js#L194-L202

Just out of curiosity, what would be the benefit/your usecase for changing the PR title and description?

**Reply by dhruvkb** [reply]
I suppose we can use some templating syntax in the supplied string and the body can be generated by filling in valid placeholders, eg:
```js
const body = PR_DESCRIPTION
body = body
	.replace('{{ github_repo }}', GITHUB_REPOSITORY)
	.replace('{{ changed_files }}', changedFiles)
```

This allows the supplied placeholder to use some `{{ changed_files }}` for changed file, `{{ github_repo }}` for repo name and so on.

I wanted to add more information about the nature of the PR to the description. Since our action runs every night, it will be force-pushing the branch multiple times as long as it is not merged. So I wanted to add the reason for the force push in the description.

**Reply by BetaHuhn** [reply]
For your usecase, would a `PR_BODY` option work instead? It would be inserted into the already present PR body. Maybe above the changed files:

```js
const body = dedent(` 
   Synced local file(s) with [${ GITHUB_REPOSITORY }](https://github.com/${ GITHUB_REPOSITORY }). 

   ${ PR_BODY } 
  
   ${ changedFiles } 
  
   --- 
  
   This PR was created automatically by the [repo-file-sync-action](https://github.com/BetaHuhn/repo-file-sync-action) workflow run [#${ process.env.GITHUB_RUN_ID || 0 }](https://github.com/${ GITHUB_REPOSITORY }/actions/runs/${ process.env.GITHUB_RUN_ID || 0 }) 
`) 
```

This would be a much smaller change to the existing code base and still gives you the option to *customize* the PR body.

I like to keep this action as simple as possible, both from a code perspective and from a user's perspective.

**Reply by dhruvkb** [reply]
@BetaHuhn if you're aiming for code minimalism, I'm completely okay with not having the entire feature at all. It was just something I felt might be good-to-have, not even remotely necessary or deal-breaking.

**Reply by BetaHuhn** [reply]
Okay, no problem! I still see the use case to at least add your own context to the PR, so a `PR_BODY` option might be a good addition.

Would you be willing to create a PR for this @dhruvkb? If not, totally okay, then I will work on this later this week.

**Reply by dhruvkb** [reply]
Sure, I could do that 👍! I'll create a PR shortly.

**Reply by BetaHuhn** [reply]
Awesome!

**Reply by dhruvkb** [reply]
Done, please see #80.

**Reply by BetaHuhn** [reply]
Great, will take a look!

**Comment by BetaHuhn** [comment, marked as answer]
:tada: This feature is included in version [v1.10.0](https://github.com/BetaHuhn/repo-file-sync-action/releases/tag/v1.10.0) :tada:

To use the latest version, use the Action with the `v1` tag:

```yml
uses: BetaHuhn/repo-file-sync-action@v1
```

Or the `latest` tag:

```yml
uses: BetaHuhn/repo-file-sync-action@latest
```
